core/domain/interaction_validation_jobs_one_off_test.py

```diff
@@ -823,6 +823,122 @@ def test_for_customization_arg_validation_job(self):
+            'html': (
+                '<oppia-noninteractive-image alt-with-value="&amp;quot;A '
+                'circle divided into equal fifths.&amp;quot;" '
+                'caption-with-value="&amp;quot;Hello&amp;quot;" '
+                'filepath-with-value="&amp;quot;xy.z.png&amp;quot;">'
+                '</oppia-noninteractive-image>'
+            )
+        }
+
+        exploration2 = exp_domain.Exploration.create_default_exploration(
+            self.NEW_EXP_ID, title='title', category='category')
+        exploration2.add_states(['State1', 'State2'])
+        exp2_state1 = exploration2.states['State1']
+        exp2_default_outcome1 = state_domain.Outcome(
+            'State2',
+            state_domain.SubtitledHtml(
+                'default_outcome',
+                (
+                    '<p><oppia-noninteractive-link text-with-value="'
+                    '&amp;quot;Test link?&amp;quot;" url-with-'
+                    'value="&amp;quot;test.com&amp'
+                    ';quot;"></oppia-noninteractive-link></p>'
+                )
+            ), False, [], None, None
+        )
+
+        with self.swap(state_domain.SubtitledHtml, 'validate', mock_validate):
+            exp1_state1.update_content(
+                state_domain.SubtitledHtml.from_dict(exp1_content1_dict))
+            exp1_state2.update_interaction_default_outcome(
+                exp1_default_outcome2)
+            exp1_state3.update_content(
+                state_domain.SubtitledHtml.from_dict(exp1_content3_dict))
+            exp_services.save_new_exploration(self.albert_id, exploration1)
+
+            exp2_state1.update_interaction_default_outcome(
+                exp2_default_outcome1)
+            exp_services.save_new_exploration(self.albert_id, exploration2)
+
+            # Start CustomizationArgsValidation job on sample exploration.
+            job_id = (
+                interaction_validation_jobs_one_off
+                .InteractionRTECustomizationArgsValidationOneOffJob.create_new(
+                    ))
+            (
+                interaction_validation_jobs_one_off
+                .InteractionRTECustomizationArgsValidationOneOffJob.enqueue(
+                    job_id))
+            self.process_and_flush_pending_tasks()
+
+        actual_output = (
+            interaction_validation_jobs_one_off
+            .InteractionRTECustomizationArgsValidationOneOffJob.get_output(
+                job_id))
+
+        expected_output = [(
+            '[u"Invalid URL: Sanitized URL should start with \'http://\' or '
+            '\'https://\'; received htt://link.com", '
+            '[u\'<p><oppia-noninteractive-link text-with-value="&amp;quot;'
+            'What is a link?&amp;quot;" url-with-value="&amp;quot;htt://'
+            'link.com&amp;quot;"></oppia-noninteractive-link></p>\', '
+            'u\'Exp Id: exp_id0\']]'
+        ), (
+            '[u"Invalid URL: Sanitized URL should start with \'http://\' '
```

Comment:
This is what I mean -- shouldn't the first two be grouped? Otherwise you'll end up with several thousand entries all exhibiting the same error.

Reply:
So, I checked how the errors are being raised for validation. It is fine for errors being raised directly in [components.py](https://github.com/oppia/oppia/blob/develop/extensions/rich_text_components/components.py) since they only have an error message and do not flag the incorrect part in the message (that part will already be obtained in the values of the dict produced by the job).

For errors which are raised after validation in [schema_utils.py](https://github.com/oppia/oppia/blob/develop/schema_utils.py#L252), we have this issue since the errors include the incorrect part which was received. I am not sure if we should modify that since it is required for debugging errors when we are not running collecting them through a job.

Reply:
Ah I see. I don't propose to modify that part (the error message) at all. I'm just saying that, if you include the Exp ID in the first part of the yield from map(), you're going to not group errors from different explorations even if they are the same in all other respects. 

Is that what you want? There are several thousand explorations on the server, so this might not be as useful as you might like (e.g. if the same error is output for a whole bunch of different explorations and then the output gets truncated, then the run would "discover" just one error). You might even consider truncating the output per unique error message after a certain point, so that you can at least see all the errors.

Reply:
Yes, you are right - it would be difficult to handle messages per exp. Updated the output generation to group by error category.